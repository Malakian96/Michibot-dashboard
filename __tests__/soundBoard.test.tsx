import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SoundBoard from '../src/components/Soundboard';

describe('SoundBoard', () => {
  const audios = ['audio1.mp3', 'audio2.mp3'];
  const handlePlaySound = jest.fn();

  it('renders correctly with given props', () => {
    const { getByText } = render(
      <SoundBoard
        audios={audios}
        handlePlaySound={handlePlaySound}
        isButtonDisabled={false}
      />,
    );

    audios.forEach((audio) => {
      expect(getByText(audio)).toBeInTheDocument();
    });
  });

  it('disables buttons when isButtonDisabled is true', () => {
    const { getByTestId } = render(
      <SoundBoard
        audios={audios}
        handlePlaySound={handlePlaySound}
        isButtonDisabled={true}
      />,
    );

    audios.forEach((audio) => {
      expect(getByTestId(`sound-button${audio}`)).toBeDisabled();
    });
  });

  it('calls handlePlaySound when a button is clicked', () => {
    const { getByTestId } = render(
      <SoundBoard
        audios={audios}
        handlePlaySound={handlePlaySound}
        isButtonDisabled={false}
      />,
    );

    fireEvent.click(getByTestId(`sound-button${audios[0]}`));
    expect(handlePlaySound).toHaveBeenCalledWith(audios[0]);
  });

  it('matches snapshot', () => {
    const { container } = render(
      <SoundBoard
        audios={audios}
        handlePlaySound={handlePlaySound}
        isButtonDisabled={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
