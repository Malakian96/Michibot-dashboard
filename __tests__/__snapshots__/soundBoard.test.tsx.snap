// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoundBoard matches snapshot 1`] = `
<div>
  <div
    class="mt-4 grid grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4"
  >
    <div
      class="flex flex-col items-center"
    >
      <button
        class="w-32 h-32 rounded-md overflow-hidden hover:scale-105 transition-transform"
        data-testid="sound-buttonaudio1.mp3"
        style="background-image: url(https://media.giphy.com/media/JIX9t2j0ZTN9S/giphy.gif); background-size: cover; background-position: center;"
      />
      <span
        class="text-xs text-white mt-1"
      >
        audio1.mp3
      </span>
    </div>
    <div
      class="flex flex-col items-center"
    >
      <button
        class="w-32 h-32 rounded-md overflow-hidden hover:scale-105 transition-transform"
        data-testid="sound-buttonaudio2.mp3"
        style="background-image: url(https://media.giphy.com/media/mlvseq9yvZhba/giphy.gif); background-size: cover; background-position: center;"
      />
      <span
        class="text-xs text-white mt-1"
      >
        audio2.mp3
      </span>
    </div>
  </div>
</div>
`;
